opencga-analysis/src/main/java/org/opencb/opencga/analysis/rga/RgaManager.java

```diff
@@ -1952,4 +1991,57 @@ public Preprocess setEvent(Event event) {
             return this;
         }
     }
+
+    /*
+    CACHE METHODS
+     */
+    private String generateCacheKey(String method, String studyStr, Query query, QueryOptions options) {
+        ObjectMap map = new ObjectMap()
+                .append("method", method)
+                .append("study", studyStr);
+        if (query != null) {
+            map.putAll(query);
+        }
+        if (options != null) {
+            map.putAll(options);
+        }
+        // Sort the keys
+        List<String> sortedKeys = map.keySet().stream().sorted().collect(Collectors.toList());
+        List<String> queryList = new ArrayList<>(map.size());
+        for (String key : sortedKeys) {
+            queryList.add(key + "=" + map.get(key));
+        }
+        return DigestUtils.sha256Hex(StringUtils.join(queryList, ";"));
+    }
+
+    private void cacheResults(String method, String studyStr, Query query, QueryOptions options, OpenCGAResult<?> result) {
+        if (!storageConfiguration.getRga().isCache()) {
```

Comment:
It'd be nice to ensure that the cache is disabled when building the aux collection, right?

Reply:
That's not what we should be controlling in my opinion. When the aux collection is being built we should not accept queries.

Reply:
This is an interesting point, I have too many comments to discuss here. This is _a must-have_ certainly, but not a big or blocker issue at the moment.